src/Select.js

```diff
@@ -1,238 +1,215 @@
-        height: 100%;
-        outline: 0;
-        /** 15px => 16px inner spacing - 1px of the border**/
-        padding: 0 12px 0 15px;
-        width: 100%;
-        -webkit-appearance: none;
-        -moz-appearance: none;
-    }
+import React, { Component } from 'react'
+import propTypes from 'prop-types'
+import {
+    statusPropType,
+    singleSelectedPropType,
+    multiSelectedPropType,
+} from './common-prop-types.js'
+import { InputWrapper } from './Select/InputWrapper.js'
+import { MenuWrapper } from './Select/MenuWrapper.js'
+
+// Keycodes for the keypress event handlers
+const ESCAPE_KEY = 27
+const SPACE_KEY = 32
+const UP_KEY = 38
+const DOWN_KEY = 40
 
-    .select {
-        align-items: center;
-        box-sizing: border-box;
-        display: flex;
-        height: 42px;
-        left: 1px;
-        position: relative;
-        border: 1px solid ${colors.grey500};
-        border-radius: 3px;
-        box-shadow: inset 0 0 0 1px rgba(102, 113, 123, 0.15),
-            inset 0 1px 2px 0 rgba(102, 113, 123, 0.1);
-        font-size: 14px;
+export class Select extends Component {
+    state = {
+        open: false,
     }
 
-    option:not(:checked) {
-        color: black;
-    }
+    selectRef = React.createRef()
+    inputRef = React.createRef()
+    menuRef = React.createRef()
 
-    select:-moz-focusring {
-        color: transparent;
-        text-shadow: 0 0 0 #000;
-    }
+    componentDidMount() {
+        if (this.props.initialFocus) {
+            this.inputRef.current.focus()
+        }
 
-    .select.dense {
-        height: 34px;
+        document.addEventListener('click', this.onOutsideClick)
     }
 
-    .select.focus {
-        border-color: ${colors.teal400};
+    componentWillUnmount() {
+        document.removeEventListener('click', this.onOutsideClick)
     }
 
-    .select.valid {
-        border-color: ${theme.valid};
-    }
+    onFocus = e => {
+        const { onFocus, disabled } = this.props
 
-    .select.warning {
-        border-color: ${theme.warning};
-    }
+        if (disabled) {
+            return
+        }
 
-    .select.error {
-        border-color: ${theme.error};
+        if (onFocus) {
+            onFocus(e)
+        }
     }
 
-    .disabled,
-    .disabled-select {
-        border-color: ${theme.disabled};
-        background-color: ${colors.grey100};
-        color: ${theme.disabled};
-        cursor: not-allowed;
-    }
+    onToggle = e => {
+        if (this.props.disabled) {
+            return
+        }
 
-    .status-icon {
-        flex-shrink: 0;
-        width: 24px;
-        height: 24px;
-        margin-right: ${spacers.dp4};
-        margin-left: ${spacers.dp12};
-    }
+        e.stopPropagation()
 
-    .status-icon:empty {
-        display: none;
+        this.setState(prevState => ({ open: !prevState.open }))
     }
 
-    .status-icon:last-child {
-        margin-right: ${spacers.dp32};
-    }
-`
+    onSingleSelectSelection = e => {
```

Comment:
The name implies that this component is connected to the SingleSelect to some degree.
This should rather be something like `closeMenu` which then can be used by the `SingleSelect` (or any other component that's created with the `Select` that wants to do that)

Reply:
I also think that this requires proper documentation as this can't be reflected with prop types

Reply:
I still think it's bad design to couple these things in a bi-directional way. The exception is not where the code introduces the exception (it's the `SingleSelect` that requires this behavior, not the `Select`)

With the solution described above (the `focus` state being controlled), the `SingleSelect` would call `closeMenu` and `props.onBlur` without having to have this information in the `Select`.

Reply:
Resolved, will be addressed in another PR when lifting up the controllable focus state

Reply:
I've addressed it here 6e9350160e534ddc071b3ad4df9638be5dd6a3ea